### DISCUSSION TITLE
[11 gennaio 2022 - B] Esercizio 2

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/psd2223#200] [Esami]
**Esercizio 2 (2+2+1+2 punti)**

- Si consideri il circuito in figura e si scriva l’espressione della funzione _f_
- Trasformare tale espressione, usando assiomi e regole dell’algebra di Boole, in forma normale SOP
- Si stenda la tavola di verità di _f_
- Si scrivano le espressioni minimali SOP e POS di _f_

![image](https://user-images.githubusercontent.com/69389461/211821989-b4d4608a-f203-44f6-9be4-4a3b3e466766.png)

**Comment by CiottoloMaggico** [comment]
![immagine](https://user-images.githubusercontent.com/79638284/216760748-43234c14-7a47-4927-a151-64efae143271.png)

**Comment by fraxb04** [comment]
@CiottoloMaggico 
è errata la tabella della verità dagli input 1000 a 1111, poiché in output avremo rispettivamente 1, 0, 0, 1, 1, 0, 0, 1.
Di conseguenza anche le sop e pos.
Di seguito allego la foto del mio esercizio:
![image](https://user-images.githubusercontent.com/122573871/216689925-fc622e81-15eb-460e-96a0-46ca5514ffa0.png)

**Comment by CiottoloMaggico** [comment]
@fraxb04 Grazie per avermelo fatto notare, effettivamente era sbagliato (ho caricato al posto del vecchio screen quello nuovo). Rifacendolo ho notato che anche tu ti sei dimenticato qualche pezzo, per esempio
Il mintermine A'BD' nella formula normale SOP manca (nella tavola lo hai inserito come 1 anche se non c'è nella formula)
Non hai contrassegnato come vero il mintermine ABC'D anche se nella formula SOP hai messo correttamente ABC', comunque oltre questo dovrebbe essere tutto giusto, salvo sviste.

**Comment by fraxb04** [comment]
@CiottoloMaggico 
Scusami nella forma SOP che ho scritto !AB!D è presente, dato che ho preso quei due mintermini nella K-Map e corrispondono alla tabella della verità…, se puoi spiegarmi meglio, grazie

**Comment by CiottoloMaggico** [comment]
![immagine](https://user-images.githubusercontent.com/79638284/216761258-09ea5fc1-1daf-4424-80d5-0896202709f0.png)
In questa parte, che presumo tu abbia fatto prima della K-map, A'BD' non è presente, nella K-map c'è poichè nella tavola della verità è contrassegnato a vero anche se nella forma SOP normale il mintermine che lo mette ad uno non c'è (dovrebbe esserci).  

Questo è il pezzo dove ti sei dimenticato di fare il prodotto tra A'B e D':
![immagine](https://user-images.githubusercontent.com/79638284/216761398-c3846d1a-bdd0-4926-b373-28440cdaefd2.png)

Il secondo errore sta nel fatto che anche se hai messo ABC' nella forma SOP normale ti sei dimenticato di mettere a vero il mintermine ABC'D nella tabella della verità.
![immagine](https://user-images.githubusercontent.com/79638284/216761464-70b013cb-bb9d-497b-a8e9-a7b0532ad921.png)

Comunque in tutto ciò penso che siano stati tutti errori di distrazione quindi non preoccuparti troppo

**Comment by fraxb04** [comment]
@CiottoloMaggico 
1) ho controllato, ma in realtà non dovrebbero corrispondere la SOP della K-map, ovvero quella minimizzata, con quella dell'esercizio 2, che sarebbe nella forma normale.
2) hai ragione, ho dimenticato di fare il prodotto, quindi basta solamente aggiungere !AB!D al risultato.
3) per fare la tavola della verità mi sono basato sul circuito, comunque hai ragione è 1 non 0 ad F, perciò occorre anche rifare la K-map sia SOP sia POS:
![image](https://user-images.githubusercontent.com/122573871/216762630-e5ea63d6-0456-4771-9b8b-59e001b62329.png)
